contributor-docs/testing.md

```diff
@@ -5,19 +5,88 @@
+- [Strategy](#strategy)
+- [Unit Tests](#unit-tests)
+  - [Overview](#overview)
+  - [What We Test](#what-we-test)
+  - [Running Tests](#running-tests)
+- [Visual Regression Tests](#visual-regression-tests)
+  - [Overview](#overview)
+  - [Prerequisites](#prerequisites)
+  - [Components](#components)
+    - [Visual Regression Testing](#visual-regression-testing)
+    - [Accessibility Verification Testing](#accessibility-verification-testing)
+- [Interaction Tests](#interaction-tests)
+  - [As A Part Of Unit Tests](#as-a-part-of-unit-tests)
+  - [Storybook Interaction Tests](#storybook-interaction-tests)
+- [Accessibility Tests](#accessibility-tests)
 - [Continous Integration](#continous-integration)
 - [FAQ](#faq)
   - [Why am I seeing `browserType.launch: Executable doesn't exist at ../path`?](#why-am-i-seeing-browsertypelaunch-executable-doesnt-exist-at-path)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 <!-- prettier-ignore-end -->
 
-## Overview
+##Strategy
+
+Strategy
+
+##Unit Tests
+
+###Overview
+
+We use [Jest](https://jestjs.io/) as our test runner to write and run our unit tests. Our prefered way of structuring tests is with `describe/it` block format and they are authored within the components's source directory with the file name of `componentName.test.tsx`
+
+We predominantly use [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro/)'s testing helpers such as
+
+- [render](https://testing-library.com/docs/react-testing-library/api#render) to render components
+- [userEvent](https://testing-library.com/docs/user-event/intro/) or [Event API](https://testing-library.com/docs/dom-testing-library/api-events/) to fire events and simulate user interaction on the component\*
+- `screen`
+- `act`
+
+To make assertions about the elements we use [Jest](https://jestjs.io/) and [jest-dom](https://github.com/testing-library/jest-dom).
+
+\*: You can read about the differences between `fireEvent` and `UserEvent` [here](https://testing-library.com/docs/user-event/intro/#differences-from-fireevent).
+
+###What We Test
+
+We try to achieve high coverage on unit tests and make sure to test components' API such as their props, events and callback functions.
+
+We also have extra set of functions to test common functionalities of components. They are
+
+- BehavesAsComponent
+- checkExports
+- checkStoriesForAxeViolations
+
+We make sure that every component has [these fundamental unit tests](https://github.com/primer/react/blob/main/src/utils/testing.tsx).
+
+###Running Unit Tests
+
+| Task                | Command                  |
+| :------------------ | :----------------------- |
+| Run unit tests      | `npm test`               |
+| Run a specific test | `npm test ComponentName` |
+| Debug unit tests    | `npm run test:watch`     |
+| Update snapshots    | `npm run test:update`    |
+| Unit test coverage  | `npm run test:coverage`  |
```

Comment:
Since these are all proxies to the Jest option, what would you think of adding the alias through `npm test`? For example:

```suggestion
| Debug unit tests    | `npm test -- --watch`     |
| Update snapshots    | `npm test -- -u`    |
```

Reply:
Hmm. I prefer running `npm run test:watch` than `npm test -- --watch` because passing params feels so imperative to me. That said, this might be a personal choice and I am happy to revert or include both if that is a preferred way  😊

Reply:
Do you usually prefer running the tools' command over package.json scripts or is here a bit of a special case do you think because of the proxies ? @joshblack

Reply:
@broccolinisoup I think if it's helpful and intuitive for folks then it's definitely worth including 👍 

I think the only things that I personally would want to avoid would be:

1. Ambiguity for an npm script, someone isn't sure what it does or its unclear what npm script to use for a task
2. An npm script for each option of a tool

For 1, I think something like `test:coverage` or `test:watch` could be confusing if there are multiple test tools (like if we're using Playwright and Jest). 

For 2, I think this is more of a personal preference on my end. I think a lot of this comes from `yarn` where you didn't have to pass `--` for options and so commands were  like `yarn test -u` and felt natural

Reply:
As we discussed in our call, great points! I updated them as you recommended 🌞

I still kept the npm scripts (like `npm run test:SOMETHING`) in the package.json for folks who are more comfortable running them over `npm test` 😊